scripts/prepare-release.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+set -e
+set -o pipefail
+
+PREVIOUS_VERSION="$(cat deploy/olm-catalog/integreatly-operator/integreatly-operator.package.yaml | grep integreatly-operator | awk -F v '{print $2}')"
+
+create_new_csv() {
+  operator-sdk generate csv --csv-version "$VERSION" --default-channel --csv-channel=rhmi --update-crds --from-version "$PREVIOUS_VERSION"
+}
+
+# Creating a new release version
+set_new_version() {
+  sed -i "s/image:.*/image: quay\.io\/integreatly\/integreatly-operator:v$VERSION/g" deploy/operator.yaml
+  sed -i "s/$PREVIOUS_VERSION/$VERSION/g" version/version.go
+  sed -i "s/$PREVIOUS_VERSION/$VERSION/g" Makefile
+  sed -i "s/PREVIOUS_TAG ?=.*/PREVIOUS_TAG ?=$PREVIOUS_VERSION/g" Makefile
+  sed -i "s/image:.*/image: quay\.io\/integreatly\/integreatly-operator:v$VERSION/g" deploy/olm-catalog/integreatly-operator/${VERSION}/integreatly-operator.v${VERSION}.clusterserviceversion.yaml
+  sed -i "s/containerImage:.*/containerImage: quay\.io\/integreatly\/integreatly-operator:v$VERSION/g" deploy/olm-catalog/integreatly-operator/${VERSION}/integreatly-operator.v${VERSION}.clusterserviceversion.yaml
+}
+
+# Creating a new tag on the already existing release
+set_new_tag() {
+  sed -i "s/image:.*/image: quay\.io\/integreatly\/integreatly-operator:v$PREVIOUS_VERSION-$TAG/g" deploy/operator.yaml
+  sed -i "s/image:.*/image: quay\.io\/integreatly\/integreatly-operator:v$PREVIOUS_VERSION-$TAG/g" deploy/olm-catalog/integreatly-operator/${PREVIOUS_VERSION}/integreatly-operator.v${PREVIOUS_VERSION}.clusterserviceversion.yaml
+  sed -i "s/version: ${PREVIOUS_VERSION}.*/version: ${PREVIOUS_VERSION}-${TAG}/g" deploy/olm-catalog/integreatly-operator/${PREVIOUS_VERSION}/integreatly-operator.v${PREVIOUS_VERSION}.clusterserviceversion.yaml
+}
+
+# Creating a new release with a tag
+set_new_version_with_tag() {
+  sed -i "s/$PREVIOUS_VERSION/$VERSION/g" Makefile
+  sed -i "s/PREVIOUS_TAG ?=.*/PREVIOUS_TAG ?=$PREVIOUS_VERSION/g" Makefile
+  sed -i "s/image:.*/image: quay\.io\/integreatly\/integreatly-operator:v$VERSION-$TAG/g" deploy/operator.yaml
```

Comment:
I'm wondering if it makes sense to actually change the image tag in deploy/operator.yaml before generating the csv. Would it not make more sense to just leave this set to master but modify the deployment in the generated csv after?

@philbrookes What do you think about this? Would it not e better to leave master pointing to the master image on quay.io?

Reply:
I've removed this for now, if @philbrookes says we should be setting the version as we were I can put it back

Reply:
@StevenTobin If we are going this route I'd update this to point to master https://github.com/integr8ly/integreatly-operator/blob/master/deploy/operator.yaml#L19

@philbrookes @maleck13 Any reason not to do this? Keep master pointing to the master quay.io image?

Reply:
I think this is probably fine. Though I have a suspicion this file is used in some e2e tests @psturc @pawelpaszki any thoughts on this?

Reply:
I'd imagine in that scenario the e2e tests should be run with --image

Reply:
Actually, the default e2e target always uses image https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L143

Reply:
ok :+1:

Reply:
We're using a pipeline library functions (in RHMI master installation & RHMI upgrade pipelines) for modifying `deploy/operator.yaml` and generating a new CSV, so it shouldn't be affected by this change.
This looks great, we could actually make use of this new script in pipelines. Thanks for letting me know about this.